src/linkedList.js

```diff
@@ -0,0 +1,131 @@
+"use strict"
+import Node from '../src/node'
+export default class LinkedList {
+  constructor(data){
+    this.head = null
+    this.tail = null
+    this.size = 0
+
+  }
+
+  getHeadNode(){
+    if (!this) {
+      return null
+    }
+    return this.head
+  }
+
+  getTailNode(){
+    if (!this) {
+      return null
+    }
+    return this.tail
+  }
+
+  contains(string){
+    let current = this.head
+    if (this.head.data == string) {
+      return true
+    }
+    current = this.head.next
+    if (current.data == string) {
+      return true
+    }
+    if (current.data !== string && current.next == null) {
+      return false
+    }
+  }
+
+
+  find(string){
+    let current = this.head
+    while(current.data !== string){
+      current = current.next
+    }
+    if (current.data == string) {
+      return current
+    }
+      return -1
+  }
+
+  insert(data){
+    const current = new Node(data)
+    this.size ++
+    if (!this.head) {
+      this.head = current
+    }
+    else  {
+      this.tail.next = current
+      this.tail = current
+    }
+    this.tail = current
+    return
+  }
+
+  insertFirst(data){
+    const current = new Node(data)
+    this.size ++
+    if (!this.head){
+      this.head = current
+      this.tail = current
+    }
+    this.head = this.head.next
+    this.head = current
+
+  }
+
+  insertBefore(string, data){
+    this.size ++
+    let stringNode = this.find(string)
+    let thisNode = this.head
+    let newNode = new Node(data)
+    if (thisNode.data == string){
+      this.head = newNode
+      this.head.next = stringNode
+      this.tail = thisNode
+    } else {
+        while (thisNode.next.data !== string) {
+          thisNode = thisNode.next
+        }
+        if (thisNode.data === string) {
+          thisNode.next = newNode
+          newNode.next.data = string
+    }
+  }
+}
+insertAfter(string, data){
+  let newNode = new Node(data)
+  this.size ++
+  let stringNode = this.find(string)
+  let thisNode = this.head
+  while (thisNode !== stringNode){
+    thisNode = thisNode.next
+  }
+  newNode.next = stringNode
+  thisNode.next = newNode
+}
+
+  clear(){
+    this.head = null
+    this.tail = null
+    this.size = 0
+    return this
+  }
+  removeFirst(){
+    this.size --
+    if (!this.head){
+      return null
+    }
+    if (!this.head.next) {
+      this.head = null
+    }
+    let newHead = this.head.next
+    this.head = newHead
+  }
+  isEmpty() {
+    if (this.head === null && this.tail === null && this.size === 0) {
```

Comment:
I did that but in one of my tests, I found that one of the nodes still existed in the tail of my linked list somehow?? Weird as hell because I thought that if you removed the head node without reassigning 'this.head' that the whole list would magically disappear...